vllm/worker/hpu_model_runner.py

```diff
@@ -317,14 +318,15 @@ def _set_block_mapping(self, metadata, batch_size, device, dtype):
         mask = mask >= metadata.block_usage.unsqueeze(-1)
         attn_bias = (torch.zeros_like(mask, dtype=dtype).masked_fill_(
             mask, -math.inf))
-        if is_fake_hpu():
-            # Unfortunately one_hot on CPU doesn't handle
-            # out of bounds classes. We need to mask those
-            # values manually
-            oob_values = metadata.block_mapping.lt(0)
-            block_mapping = metadata.block_mapping.masked_fill(oob_values, 0)
+        if is_fake_hpu() or (not htorch.utils.internal.is_lazy()
+                             and not self.enforce_eager):
+            # Unfortunately one_hot on CPU or in torch compile mode
+            # doesn't handle out of bounds classes.
+            # We need to mask those values manually
+            block_mapping = torch.nn.functional.relu(metadata.block_mapping)
             block_mapping = torch.nn.functional.one_hot(block_mapping,
                                                         num_classes=batch_size)
+            oob_values = metadata.block_mapping.lt(0)
```

Comment:
this and next line were left accidentally ?

Reply:
This and next line cannot be deleted. Negative values in metadata.block_mapping must be represented as zeros after one_hot, otherwise it will cause error in further processing. 
Here is an example for better understanding: 
metadata.block_mapping=[0, -1, -1]   --_relu()_-->   [0, 0, 0]   --_one_hot with num_classes=1_-->   [[1], [1], [1]]     --_oob value mask_-->   [[1], [0], [0]]